src/box/lua/config/utils/file.lua

```diff
@@ -55,6 +56,117 @@ local function universal_read(file_name, file_kind)
     return data
 end
 
+
+local function get_file_metadata(file_name)
+    local function escape_regex(str)
+        return str:gsub('([%.%+%-%*%?%[%]%(%)%$%^%{%}%|%<%>%~])', '%%%1')
+    end
+
+    -- Tags are only supported for Lua files.
+    if type(file_name) ~= 'string' or not file_name:endswith('.lua') then
+        error(('Invalid file name %q'):format(file_name), 0)
+    end
+
+    local file, err = io.open(file_name, "r")
+    if file == nil then
+        error(('Unable to open %q: %s'):format(file_name, err), 0)
+    end
+
+    local metadata = {}
+
+    local line_count = 0
+    local yaml_string = ''
+    local comment_tag = nil
+    -- Currently the only supported version is 1 and its value is not used.
+    -- luacheck: ignore 311 value assigned to a local variable is unused.
+    local metadata_version = nil
+    local in_comment, in_yaml = false, false
+    while true do
+        local line = file:read("*line")
+        if line == nil then
+            break
+        end
+        line_count = line_count + 1
+
+        -- Skip shebang.
+        if line_count == 1 and line:startswith('#!') then
+            goto next_line
+        end
+
+        -- Skip empty lines.
+        line = line:lstrip()
+        if line == '' then
+            goto next_line
+        end
+
+        if not in_comment and line:startswith('--') then
+            in_comment = true
+        end
+
+        if in_comment then
+            if not line:startswith('--') then
+                break
+            end
+        end
+
+        if in_comment and not in_yaml then
+            comment_tag, metadata_version = line:match(
+                '^(%-%-[%s]+)%-%-%- #tarantool.metadata.v([%d]+)[%s]*$')
+            metadata_version = tonumber(metadata_version)
+
+            if comment_tag ~= nil and metadata_version ~= nil then
+                comment_tag = escape_regex(comment_tag)
+                in_yaml = true
+            end
+        end
+
+        if in_yaml then
+            line = line:match(('^%s(.*)'):format(comment_tag))
```

Comment:
Nit: It looks simpler to just strip N chars from the beginning (and determine this N by the document start line).

Reply:
It would be simpler, but a lot less strict and could lead to some problems in future with files like:
```lua
--  #tarantool.metadata.v1
--a foo: bar
--bb...
```

the 'a' and 'bb' on lines 2 and 3 will just be ignored, and I'm not sure we want to allow that.